core/config/project_settings.cpp

```diff
@@ -1124,8 +1124,12 @@ void ProjectSettings::set_setting(const String &p_setting, const Variant &p_valu
 	set(p_setting, p_value);
 }
 
-Variant ProjectSettings::get_setting(const String &p_setting) const {
-	return get(p_setting);
+Variant ProjectSettings::get_setting(const String &p_setting, const Variant &p_default_value) const {
+	Variant v = get(p_setting);
+	if (v.get_type() == Variant::NIL) {
+		return p_default_value;
```

Comment:
Shouldn't we use `has_setting()` or check `props.has()` directly? Technically, being null and not existing is not the same thing. Can't say if there are settings that benefit from it, but there might be.

Reply:
Yes you are right. This is more safe and also makes more sense from the developer perspective, e.g. if `has_setting` returns false, I expect the `default_value` to be returned when using `get_setting`. Thanks!

Reply:
One more note: `get_setting()` use the `get()` method in `object.cpp` instead of querying the `props` collection.
I think that's why I didn't used `has_setting` in the past, although it looks more natural.
Am I missing something here?

Reply:
When you go through `get` we also convert the name if a feature override for it is available. I don't think it should affect `has_setting`.

Reply:
Makes sense, yeah. But the `get()` method is still doing a bit more, no? 
e.g. Line 353: `const Variant *const *V = metadata_properties.getptr(p_name);`
And this is not included when do a `has_setting()` check. 
Although I would not expect this to make any problems, still a bit weird.

Reply:
You should look at `ProjectSettings::_get`, not at the low-level `Object` implementation. The `Object` one indeed does a lot more, but it's a general accessor for all class properties. Here we're only interested in settings, which should correctly validate with `has_setting`.

Reply:
Alright, thanks for the clarification. :)